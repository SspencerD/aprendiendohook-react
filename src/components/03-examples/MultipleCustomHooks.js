import React from 'react'
import { useCounter } from '../../Hooks/useCounter';
import { useFetch } from '../../Hooks/useFetch';
import '../02-useEffect/effects.css';

export const MultipleCustomHooks = () => {


    const {counter,increment}= useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);


    const { author, quote } = !!data && data[0];

    return (
        <div>

            <h1>Personajes de BreakingBad</h1>
            <hr />

            { loading ? (
                <div className="alert alert-warning">
                    Cargando...
                </div>

            )
                :
                (
                    <figure className="border-aqua border-1">
                        <blockquote className="blockquote text-right">
                            <p className='mb-0'>{quote}</p>
                        </blockquote>
                        <figcaption className="blockquote-footer">
                            <p className='mb-1'>{author}</p>
                        </figcaption>
                    </figure>
                )



            }

            <button className="btn btn-outline-info" onClick={increment}> Siguiente Autor</button>



        </div>
    )
}
