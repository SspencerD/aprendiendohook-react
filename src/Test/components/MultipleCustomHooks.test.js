import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../Hooks/useFetch';
import { useCounter } from '../../Hooks/useCounter';
jest.mock('../../Hooks/useFetch');
jest.mock('../../Hooks/useCounter');


describe('Pruebas en <MultipleCustomHooks/>', () => {


    beforeEach( () => {

        useCounter.mockReturnValue({
            counter: 1,
            increment: ()=>{}
        });
        

    });

    test('debe de mostrarse correctamente', () => {
        
            useFetch.mockReturnValue({

                data:null,
                loading: true,
                error: null
            
        });
        
        const wrapper= shallow(<MultipleCustomHooks/>)

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar info ', () => {

        useFetch.mockReturnValue({

            data:[{
                author:' Goku',
                quote: 'Que waaa'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks/>);

        expect( wrapper.find('.alert').exists()).toBe(false);
        expect( wrapper.find('.mb-0').text().trim()).toBe('Que waaa');
        expect( wrapper.find('figcaption').exists()).toBe(true);


        
    });
});