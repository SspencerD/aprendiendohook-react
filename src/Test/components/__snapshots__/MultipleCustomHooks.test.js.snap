// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Personajes de BreakingBad
  </h1>
  <hr />
  <div
    className="alert alert-warning"
  >
    Cargando...
  </div>
  <button
    className="btn btn-outline-info"
    onClick={[Function]}
  >
     Siguiente Autor
  </button>
</div>
`;
